darts/models/forecasting/forecasting_model.py

```diff
@@ -241,6 +260,230 @@ def min_train_series_length(self) -> int:
+        """
+        (
+            min_target_lag,
+            max_target_lag,
+            min_past_cov_lag,
+            min_future_cov_lag,
+            max_future_cov_lag,
+        ) = self.extreme_lags
+
+        return max(
+            max_target_lag,
+            max_future_cov_lag if max_future_cov_lag else 0,
+        ) - min(
+            min_target_lag if min_target_lag else 0,
+            min_past_cov_lag if min_past_cov_lag else 0,
+            min_future_cov_lag if min_future_cov_lag else 0,
+        )
+
+    @property
+    def predict_sample_time_index_length(self) -> int:
+        """
+        Class property defining the required time_index length for one predict sample, for any model.
+        A predict sample is the minimum required set of series and covariates chunks to be able to predict
+        a single point.
+        """
+        (
+            min_target_lag,
+            max_target_lag,
+            min_past_cov_lag,
+            min_future_cov_lag,
+            max_future_cov_lag,
+        ) = self.extreme_lags
+
+        return (
+            max_future_cov_lag
+            if max_future_cov_lag
+            else 0
+            - min(
+                min_target_lag if min_target_lag else 0,
+                min_past_cov_lag if min_past_cov_lag else 0,
+                min_future_cov_lag if min_future_cov_lag else 0,
+            )
+        )
+
+    def _get_historical_forecastable_time_index(
+        self,
+        series: Optional[TimeSeries] = None,
+        past_covariates: Optional[TimeSeries] = None,
+        future_covariates: Optional[TimeSeries] = None,
+        is_training: Optional[bool] = False,
+    ) -> Union[pd.DatetimeIndex, pd.RangeIndex, None]:
+        """
+        Private function that returns the largest time_index representing the subset of each timestamps
+        for which historical forecasts can be made, given the model's properties, the training series
+        and the covariates.
+            - If ``None`` is returned, there is no point where a forecast can be made.
+
+            - If ``is_training=False``, it returns the time_index subset of predictable timestamps.
+
+            - If ``is_training=True``, it returns the time_index subset of trainable timestamps. A trainable
+            timestamp is a timestamp that have at least a training sample of length ``self.training_sample_length``
+             preceding it.
+
+
+        Parameters
+        ----------
+        series
+            Optionally, a target series.
+        past_covariates
+            Optionally, a past covariates.
+        future_covariates
+            Optionally, a future covariates.
+        is_training
+            Whether the returned time_index should be taking into account the training.
+
+        Returns
+        -------
+        Union[pd.DatetimeIndex, pd.RangeIndex, None]
+            The longest time_index that can be used for historical forecasting.
+
+        Examples
```

Comment:
+1 for the example (unfortunately the whole docstring won't be rendered for private methods, but still useful)